src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx

```diff
@@ -168,10 +168,42 @@ export class DataSourceSelectable extends React.Component<
     }
   }
 
+  getLable(): string {
+    const selectedOption =
+      this.state.selectedOption &&
+      this.state.selectedOption.length > 0 &&
+      this.state.selectedOption[0]
+        ? this.state.selectedOption[0]
+        : '';
+    if (!selectedOption) return '';
+    else {
+      if (selectedOption.id === '' || (selectedOption.id && selectedOption.label)) {
+        return selectedOption.label;
+      } else {
+        // label not exist, get it from dataSourceOptions
+        const options = this.state.dataSourceOptions;
+        if (options && options.length > 0) {
+          // if dataSourceOptions is not empty
+          const mappingOption = options.find((option) => selectedOption.id! === option.id);
+          if (mappingOption && mappingOption.label) {
+            return mappingOption.label;
```

Comment:
Could we use optional chain operator

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining